### ISSUE TITLE
`Dispatcher.InvokeAsync` overloads are confusing

### ISSUE THREAD

**Issue by tomenscape** [AvaloniaUI/Avalonia#11211]
**Describe the bug**
Here are two dispatcher invoke calls. What's the difference between them?

```C#

Task MyAsyncMethod() => Task.Delay(500);

await Dispatcher.UIThread.InvokeAsync(MyAsyncMethod);
await Dispatcher.UIThread.InvokeAsync(MyAsyncMethod, priority: default);
```

Answer: the first line waits until the dispatcher has invoked `MyAsyncMethod`. The second line waits until the dispatcher has invoked `MyAsyncMethod` _and the Task it returns has completed_.

This very important difference is not clear at all from the method names. The change in behaviour being caused by the introduction of a dispatcher priority is also entirely unexpected.

**Expected behavior**
The `InvokeAsync` overloads which unwrap the result of a `Func<Task>` method should have different names, e.g. `InvokeAsyncAndUnwrap`.

**Comment by kekekeks** [comment]
`Dispatcher.UIThread.InvokeAsync<Task>(MyAsyncMethod)`

**Comment by tomenscape** [comment]
Fixed by #11214.
